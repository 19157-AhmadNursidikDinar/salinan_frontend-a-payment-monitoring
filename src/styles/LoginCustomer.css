@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@200&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@500&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;700&display=swap");

.container {
  display: flex;
  flex-direction: row;
  width: 100%;
  min-height: 100vh;
  background-color: grey;
}
.left-side {
  flex: 1;
  display: flex;
  width: 50%;
  justify-content: center;
  align-items: center;
  background-color: white;
}
.flex-center {
  display: flex;
  justify-content: center;
}
.right-side {
  flex: 1;
  width: 50%;
  background-color: #e6f2ff;
  padding-bottom: 10%;
  padding-left: 5%;
}
.box-cont {
  display: block;
  width: 70%;
  padding-bottom: 50px;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 2px 4px #676565;
}
.header-logo {
  display: flex;
  margin-top: 30px;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: 20px;
}
.inside-box {
  width: 100%;
  padding-left: 20%;
  padding-right: 20%;
  display: flex;
  justify-content: center;
  flex-direction: column;
}
.head-logo {
  width: 79px;
  height: 70px;
}
.title-section {
  line-height: 37px;
  margin-left: 9px;
  font-family: "Poppins", sans-serif;
  font-family: 25px;
  font-size: 30px;
  font-weight: bold;
  color: #444444;
}
.title-section span {
  color: #21afd3;
}
.txt-signIn {
  line-height: 20px;
  font-family: "Poppins", sans-serif;
  font-size: 13px;
  font-weight: bold;
  color: #444444;
  text-align: center;
}
.user-pass {
  width: 100%;
  background-color: #ffffff;
  text-align: center;
}
:placeholder {
  color: #676565;
  font-family: "Roboto", sans-serif;
  font-size: 1px;
}
.wrapped-txtfield {
  display: flex;
  padding: 8px 10px;
  margin-bottom: 12px;
  margin-top: 4px;
  margin-left: auto;
  margin-right: auto;
}
.txtfield {
  box-sizing: border-box;
  text-align: center;
  border: none;
  outline: none;
  border-bottom: 1.5px solid #3da5c6;
}
.wrapped-rememberMe {
  display: flex;
  padding: 8px 10px;
  margin-bottom: 12px;
  flex-direction: row;
  align-items: center;
  color: #444444;
  font-weight: bold;
}

.wrapped-signin {
  display: flex;
  padding: 0px 20px;
}
.btn-signIn {
  display: flex;
  padding-top: 5px;
  padding-bottom: 5px;
  justify-content: center;
  align-items: center;
  border-radius: 10px !important;
  background: linear-gradient(45deg, #15aad9 30%, #7ee0ff 90%);
}
.btn-signIn b {
  color: #ffff;
  font-size: 15px;
  font-family: poppins;
  font-weight: 500;
}
.welcome-title {
  flex: 1;
  padding-left: 61px;
  padding-top: 106px;
  width: 100%;
  padding-bottom: 48px;
}
.welcome {
  font-family: "Roboto", sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 40px;
  line-height: 30px;
  margin-bottom: 0.4em;
  color: #444444;
}
.text-notice {
  font-family: "Poppins", sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 30px;
  line-height: 35px;
  color: #444444;
}
.img {
  flex: 2;
  width: 100%;
}
.circled-image {
  width: 430px;
  height: 420px;
  margin-left: 120px;
  resize: block;
}
.banker-image {
  width: 450px;
  height: 424px;
  position: absolute;
  top: 25%;
  left: 55%;
}

.switch-signin{
  color: #15aad9;
}