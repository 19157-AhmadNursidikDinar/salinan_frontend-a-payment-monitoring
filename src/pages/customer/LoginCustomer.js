import React, { useState } from "react";
import { Link } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import { blue } from "@material-ui/core/colors";
import TextField from "@material-ui/core/TextField";
import Alert from "@material-ui/lab/Alert";
import InputAdornment from "@material-ui/core/InputAdornment";
//import @material-ui/icons for icon username, password dan checkbox rememberMe
import PersonIcon from "@material-ui/icons/Person";
import LockIcon from "@material-ui/icons/Lock";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Checkbox from "@material-ui/core/Checkbox";
import Button from "@material-ui/core/Button";
//import css and assets
import Logo from "../../assets/logoPM.png";
import BankerLogo from "../../assets/Banker.png";
import Circle from "../../assets/elipse.png";
import { useFormik } from "formik";
import * as Yup from "yup";
import AuthService from "../../services/auth.service";
import "../../styles/LoginCustomer.css";

const BlueCheckbox = withStyles({
  root: {
    color: blue[300],
    "&$checked": {
      color: blue[600],
    },
  },
  checked: {},
})((props) => <Checkbox color="default" {...props} />);

const validationSchema = Yup.object().shape({
  username: Yup.string().required("Insert username!"),
  password: Yup.string()
    .min(6, "Use combination of 6 character or more")
    .required("Insert Password"),
  rememberMe: Yup.boolean(),
});

export default function LoginCustomer(props) {
  const [errorMsg, setErrorMsg] = useState("");
  const formik = useFormik({
    initialValues: {
      username: "",
      password: "",
      rememberMe: false,
    },
    validationSchema: validationSchema,
    validateOnBlur: false,
    validateOnChange: false,
    onSubmit: async ({ username, password, rememberMe }) => {
      // console.log(values);
      const result = await AuthService.login({
        username,
        password,
        rememberMe,
      });
      console.log({ result });
      if (!Boolean(result.error)) {
        props.history.push("/customer");
      } else {
        setErrorMsg(result.error.response.data.msg);
      }
    },
  });

  return (
    <div className="container">
      <div className="left-side">
        <div className="box-cont">
          <div className="flex-center">
            <div className="header-logo">
              <img src={Logo} alt="app-logo" className="head-logo" />
              <h1 className="title-section">
                Pay<span>ment</span> <br /> Monitoring
              </h1>
            </div>
          </div>
          <h3 className="txt-signIn">Sign in to continue our application</h3>
          <div className="inside-box">
            {errorMsg && (
              <Alert severity="error" style={{ margin: "0.5em 0" }}>
                {errorMsg}
              </Alert>
            )}
            <form onSubmit={formik.handleSubmit}>
              <div className="user-pass">
                <div className="wrapped-txtfield">
                  <TextField
                    className="txtfield"
                    type="text"
                    placeholder="Username"
                    name="username"
                    value={formik.values.username}
                    onChange={formik.handleChange}
                    variant="outlined"
                    fullWidth
                    InputProps={{
                      startAdornment: (
                        <InputAdornment position="start">
                          <PersonIcon style={{ color: "#2EB7E2" }} />
                        </InputAdornment>
                      ),
                    }}
                    disabled={formik.isSubmitting}
                    error={
                      Boolean(formik.errors.username) && formik.touched.username
                    }
                    helperText={formik.errors.username}
                  />
                </div>
                <div className="wrapped-txtfield">
                  <TextField
                    name="password"
                    className="txtfield"
                    type="password"
                    placeholder="Password"
                    variant="outlined"
                    fullWidth
                    InputProps={{
                      startAdornment: (
                        <InputAdornment position="start">
                          <LockIcon style={{ color: "#2EB7E2" }} />
                        </InputAdornment>
                      ),
                    }}
                    value={formik.values.password}
                    onChange={formik.handleChange}
                    disabled={formik.isSubmitting}
                    error={
                      Boolean(formik.errors.password) && formik.touched.password
                    }
                    helperText={formik.errors.password}
                  />
                </div>
                <div className="wrapped-rememberMe">
                  <FormControlLabel
                    control={
                      <BlueCheckbox
                        name="rememberMe"
                        checked={formik.values.rememberMe}
                        onChange={formik.handleChange}
                        disabled={formik.isSubmitting}
                      />
                    }
                    label="Remember Me"
                  />
                </div>
                <div className="wrapped-signin">
                  <Button className="btn-signIn" fullWidth type="submit">
                    <b>{formik.isSubmitting ? "Loading..." : "Sign In"}</b>
                  </Button>
                </div>
                <div
                  style={{
                    marginTop: 20,
                    justifyContent: "center",
                    display: "flex",
                  }}
                >
                  <Link className="switch-signin" to="/login-officer">
                    Sign In as Officer
                  </Link>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div className="right-side">
        <div className="welcome-title">
          <h1 className="welcome">WELCOME BACK,</h1>
          <h2 className="text-notice">Make your payment efficiently</h2>
        </div>
        <div className="img">
          <img className="circled-image" src={Circle} alt="Round.png" />
          <img className="banker-image" src={BankerLogo} alt="BankerLogo" />
        </div>
      </div>
    </div>
  );
}
