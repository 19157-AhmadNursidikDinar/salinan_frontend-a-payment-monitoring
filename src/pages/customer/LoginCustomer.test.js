import React from "react";
import { shallow } from "enzyme";
import LoginCustomer from "./LoginCustomer";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";

describe("Login Customer", () => {
    const wrapper = shallow(<LoginCustomer />);
    it("should render Login Customer", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("should render two TextField", () => {
        expect(wrapper.find(TextField).length).toBe(2);
    });
    it("should render one Button", () => {
        expect(wrapper.find(Button).length).toBe(1);
    });
});