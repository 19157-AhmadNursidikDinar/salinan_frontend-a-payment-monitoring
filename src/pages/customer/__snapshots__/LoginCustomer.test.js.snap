// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Customer should render Login Customer 1`] = `ShallowWrapper {}`;
